akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -390,18 +398,37 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       becomeUninitialized()
       join(address)
     case JoinSeedNodes(newSeedNodes) ⇒
+      resetJoinSeedNodesDeadline()
       becomeUninitialized()
       joinSeedNodes(newSeedNodes)
     case msg: SubscriptionMessage ⇒ publisher forward msg
     case _: Tick ⇒
-      if (deadline.exists(_.isOverdue)) {
+      if (joinSeedNodesDeadline.exists(_.isOverdue))
+        joinSeedNodesWasUnsuccessful()
+      else if (deadline.exists(_.isOverdue)) {
         // join attempt failed, retry
         becomeUninitialized()
         if (seedNodes.nonEmpty) joinSeedNodes(seedNodes)
         else join(joinWith)
       }
   }: Actor.Receive).orElse(receiveExitingCompleted)
 
+  private def resetJoinSeedNodesDeadline(): Unit = {
+    joinSeedNodesDeadline = ShutdownAfterUnsuccessfulJoinSeedNodes match {
+      case d: FiniteDuration ⇒ Some(Deadline.now + d)
+      case _                 ⇒ None // off
+    }
```

Comment:
This seems a bit strange here, is `ShutdownAfterUnsuccessfulJoinSeedNodes` a nullable `Duration`? Perhaps we should make it an `Option[FiniteDuraction]` already on initialization?